src/app/components/pages/Login/index.tsx

```diff
@@ -1,44 +1,72 @@
-import React, { useState } from 'react';
+import React, { useEffect, useState } from 'react';
 import { API_URL, PAGES } from '../../../utils/contants';
+
 import './index.css';
+import background from '../../../assets/loginBackground.svg';
+
 export interface LoginProps {
   setActivePage: React.Dispatch<React.SetStateAction<PAGES>>;
 }
+
+function generateRandomToken() {
+  const randomValues = new Uint8Array(128);
+  window.crypto.getRandomValues(randomValues);
+
+  // Convert the Uint8Array to a base64 string
+  const base64String = btoa(String.fromCharCode.apply(null, randomValues));
+
+  return base64String;
+}
+
 const Login = function (props: LoginProps) {
   const {} = props;
   const [token, setToken] = useState('');
-  return (
-    <div className="login">
-      <form
-        onSubmit={async (e) => {
-          e.preventDefault();
-          const res = await fetch(API_URL + 'users/temp_token_login?token=' + token);
+
+  useEffect(() => {
+    if (token) fetchUserInfo();
+  }, [token]);
+
+  const fetchUserInfo = () => {
+    const intervalId = setInterval(async () => {
+      try {
+        const res = await fetch(`${API_URL}users/temp_token_login?token=${token}`);
+        if (res.status == 200) {
+          clearInterval(intervalId);
           const userInfo = await res.json();
           parent.postMessage({ pluginMessage: { type: 'store-user-info', userInfo: JSON.stringify(userInfo) } }, '*');
-        }}
-      >
-        <h3>Login page</h3>
-        <input
-          value={token}
-          onChange={(e) => {
-            setToken(e.target.value);
-          }}
-          required
-          placeholder="Huehive login token"
-        />
-        <button type="submit">Submit </button>
-        <div >
-          <p className='login-text'>
-            Copy login token from{' '}
-            <a href="https://huehive.co/users/figma_token" target="_blank">
-              HueHive
+        }
+      } catch {}
+    }, 4000);
```

Comment:
Let's clear the interval after 5 mins as a safety to avoid unnessesory calls to server in case user forgot to close the plugin and login.
```  setTimeout(() => {
    clearInterval(intervalId);
  }, 5 * 60 * 1000); // 5 minutes in milliseconds
```